dsgrid/config/representative_period_time_dimension_config.py

```diff
@@ -219,13 +222,23 @@ def list_expected_dataset_timestamps(self):
 class OneWeekPerMonthByHourHandler(RepresentativeTimeFormatHandlerBase):
     """Handler for format with hourly data that includes one week per month."""
 
-    def check_dataset_time_consistency(self, expected_timestamps, load_data_df):
+    def check_dataset_time_consistency(self, expected_timestamps, load_data_df, time_columns):
         logger.info("Check OneWeekPerMonthByHourHandler dataset time consistency.")
-        time_cols = self.get_timestamp_load_data_columns()
-        actual_timestamps = [
-            OneWeekPerMonthByHourType(*x.asDict().values())
-            for x in load_data_df.select(*time_cols).distinct().sort(*time_cols).collect()
-        ]
+        actual_timestamps = []
+        for row in load_data_df.select(*time_columns).distinct().sort(*time_columns).collect():
+            data = row.asDict()
+            num_none = 0
+            for val in data.values():
+                if val is None:
+                    num_none += 1
+            if num_none > 0:
+                if num_none != len(data):
```

Comment:
Feels weird to me, I also feel like I've reviewed this part before.

If there's a null, then all cols must be null, so it seems there are two correct possibilities:
- records without any null
- a single record of null
So, to me, this means as long as num_none > 1: it's an error

Reply:
What about representative time with one-week-per-month in one-table format and missing data for specific dimension combinations? All 3 time columns must be null.

Reply:
Got cha, I misinterpreted how data looks like. It's just a single row in dict form, where each item is a column.